### PR TITLE
[PLUGIN-1528] Adding Precision to set-type directive

### PR DESCRIPTION
**Pull request by minurajeeve**
Adding Precision text box in set-type directive and modifying directive format to include precision and adding corresponding test cases

## PR Type
- [x] Feature

## Links
Jira: [Jira issue #](https://cdap.atlassian.net/browse/PLUGIN-1698)

## Screenshots
image

### CODE REVIEW

File: app/cdap/components/DataPrep/Directives/ChangeDataType/DecimalOptions.tsx [hunk 1]
```diff
@@ -114,24 +115,36 @@ const useStyles = makeStyles((theme) => ({
...
   function applyDirective() {
     const option = 'decimal';
-    const extraArgs = scale !== '' ? `${scale} '${rounding || 'HALF_EVEN'}'` : '';
+    let extraArgs = scale !== '' ? `${scale}` : '';
+    if (scale !== '' || precision !== '') {
+      extraArgs += ` '${rounding || 'HALF_EVEN'}'`;
+    }
+    if (precision !== '') {
+      extraArgs = extraArgs + ` prop:{precision=${precision}}`;
+    }
```
Comment (GnsP) [inline app/cdap/components/DataPrep/Directives/ChangeDataType/DecimalOptions.tsx]: prefer template literals over string concatenation with `+` operator

**Review by GnsP — CHANGES_REQUESTED** [review]

**Comment by GnsP** [comment]
PR type should be `feature` and JIRA issue should be specified.

File: app/cdap/components/DataPrep/Directives/ChangeDataType/DecimalOptions.tsx (hunk 1 above)
Comment (minurajeeve) [reply app/cdap/components/DataPrep/Directives/ChangeDataType/DecimalOptions.tsx]: done

**Comment by minurajeeve** [comment]
> PR type should be `feature` and JIRA issue should be specified.

Done

**Review by GnsP — APPROVED** [review]

**Review by radhikav1 — APPROVED** [review]

**Pull request merged**